flow/flow.go

```diff
@@ -1040,6 +1103,176 @@ func handle(parameters interface{}, stopper chan int, report chan uint64, contex
 	}
 }
 
+func write(parameters interface{}, coreId uint8) {
+	wp := parameters.(*writeParameters)
+	IN := wp.in
+	filename := wp.filename
+
+	bufIn := make([]uintptr, 1)
+	var tempPacket *packet.Packet
+
+	f, err := os.Create(filename)
+	check(err)
+	defer f.Close()
+
+	WritePcapGlobalHdr(f)
+	for {
+		n := IN.DequeueBurst(bufIn, 1)
+		if n == 0 {
+			continue
+		}
+		tempPacket = packet.ExtractPacket(bufIn[0])
+		WritePcapOnePacket(tempPacket, f)
+	}
+}
+
+func read(parameters interface{}, coreId uint8) {
+	rp := parameters.(*readParameters)
+	OUT := rp.out
+	filename := rp.filename
+	mempool := rp.mempool
+	repcount := rp.repcount
+
+	buf := make([]uintptr, 1)
+	var tempPacket *packet.Packet
+
+	f, err := os.Open(filename)
+	check(err)
+	defer f.Close()
+
+	// Read pcap gloabl header once
+	var glHdr pcapGlobHdr
+	readPcapGlobalHdr(f, &glHdr)
+
+	count := int32(0)
+
+	for {
+		low.AllocateMbufs(buf, mempool)
+		tempPacket = packet.ExtractPacket(buf[0])
+		isEOF := readOnePacket(tempPacket, f)
+		if isEOF {
+			atomic.AddInt32(&count, 1)
+			if count == repcount {
+				break
+			}
+			_, err := f.Seek(int64(unsafe.Sizeof(glHdr)), 0)
```

Comment:
After replacing below functions to other file in packet package this will be the only "unsafe". Please change it somehow. If it is absolutely necessary, use wrapper from low.go with unsafe.

I suppose we shouldn't have any unsafe in flow.go